lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+            results['description_previous'] = pg_config.description
+            if self.description is None:
+                # need to use empty string; will be set to None by API
+                pg_spec.description = ''
+            else:
+                pg_spec.description = self.description
+
+        # Check port policies
+        results['adv_reset_at_disconnect'] = self.pp_reset
+        results['adv_block_ports'] = self.pp_block_ports
+        results['adv_traffic_shaping'] = self.pp_shaping
+        results['adv_vendor_conf'] = self.pp_vendor_conf
+        results['adv_vlan'] = self.pp_vlan
+        results['adv_uplink_teaming'] = self.pp_uplink_teaming
+        results['adv_network_rp'] = self.pp_network_rp
+        results['adv_security'] = self.pp_security
+        results['adv_netflow'] = self.pp_netflow
+        results['adv_traffic_filtering'] = self.pp_traffic_filter
+        (pg_policy_spec, changed_policy, reset_at_disconnect, block_override, shaping_override,
+         vendor_config_override, vlan_override, uplink_teaming_override, network_rp_override,
+         security_override, netflow_override, traffic_filter_override) = self.check_port_policy_config(pg_config)
+        if changed_policy:
+            changed = True
+            changed_list.append("port policies")
+            if reset_at_disconnect:
+                results['adv_reset_at_disconnect_previous'] = pg_config.policy.portConfigResetAtDisconnect
+            if block_override:
+                results['adv_block_ports_previous'] = pg_config.policy.blockOverrideAllowed
+            if shaping_override:
+                results['adv_traffic_shaping_previous'] = pg_config.policy.shapingOverrideAllowed
+            if vendor_config_override:
+                results['adv_vendor_conf_previous'] = pg_config.policy.vendorConfigOverrideAllowed
+            if vlan_override:
+                results['adv_vlan_previous'] = pg_config.policy.vlanOverrideAllowed
+            if uplink_teaming_override:
+                results['adv_uplink_teaming_previous'] = pg_config.policy.uplinkTeamingOverrideAllowed
+            if network_rp_override:
+                results['adv_network_rp_previous'] = pg_config.policy.networkResourcePoolOverrideAllowed
+            if security_override:
+                results['adv_security_previous'] = pg_config.policy.securityPolicyOverrideAllowed
+            if netflow_override:
+                results['adv_netflow_previous'] = pg_config.policy.ipfixOverrideAllowed
+            if traffic_filter_override:
+                results['adv_traffic_filtering_previous'] = pg_config.policy.trafficFilterOverrideAllowed
+            pg_spec.policy = pg_policy_spec
+
+        pg_spec.defaultPortConfig = vim.dvs.VmwareDistributedVirtualSwitch.VmwarePortConfigPolicy()
+
+        # TODO: Check Traffic shaping
+
+        # Check VLAN
+        if self.vlan_id == 0 and self.vlan_trunk_range is None and self.private_vlan_id is None:
```

Comment:
This can be simplified to `if self.vlan_id == 0`, as the other two will be `None` automatically because of the exclusivity check

Reply:
done